api.go

```diff
@@ -19,37 +19,41 @@ package rocketmq
 
 import (
 	"context"
-
+	"github.com/apache/rocketmq-client-go/consumer"
 	"github.com/apache/rocketmq-client-go/primitive"
+	"github.com/apache/rocketmq-client-go/producer"
 )
 
 type Producer interface {
 	Start() error
 	Shutdown() error
-	SendSync(context.Context, ...*primitive.Message) (primitive.SendResult, error)
-	SendAsync(context.Context, func(primitive.SendResult), ...*primitive.Message) error
-	SendOneWay(context.Context, ...*primitive.Message) error
+	SendSync(context.Context, *primitive.Message) (*primitive.SendResult, error)
+	SendAsync(context.Context, func(context.Context, *primitive.SendResult), *primitive.Message) error
+	SendOneWay(context.Context, *primitive.Message) error
 }
 
-func NewProducer(opt primitive.ProducerOptions) (Producer, error) {
-	return nil, nil
+func NewProducer(opts ...producer.Option) (Producer, error) {
+	return producer.NewDefaultProducer(opts...)
 }
 
 type PushConsumer interface {
 	Start() error
 	Shutdown() error
-	Subscribe(topic string, selector primitive.MessageSelector,
-		f func(context.Context, ...*primitive.MessageExt) (primitive.ConsumeResult, error)) error
+	Subscribe(topic string, selector consumer.MessageSelector,
+		f func(context.Context, ...*primitive.MessageExt) (consumer.ConsumeResult, error)) error
 	Unsubscribe(string) error
 }
 
+func NewPushConsumer(opts ...consumer.Option) (PushConsumer, error) {
+	return consumer.NewPushConsumer(opts...)
+}
+
 type PullConsumer interface {
 	Start() error
 	Shutdown() error
-	Pull(context.Context, string, primitive.MessageSelector, int) (primitive.PullResult, error)
-	PullFrom(context.Context, primitive.MessageQueue, int64, int) (primitive.PullResult, error)
+	Pull(context.Context, string, consumer.MessageSelector, int) (*primitive.PullResult, error)
+	PullFrom(context.Context, primitive.MessageQueue, int64, int) (*primitive.PullResult, error)
```

Comment:
primitive.MessageQueue => *primitive.MessageQueue maybe  is  betrer

Reply:
done